src/System.IO.Pipelines/src/System/IO/Pipelines/Pipe.cs

```diff
@@ -407,7 +407,7 @@ private void AdvanceReader(BufferSegment consumedSegment, int consumedIndex, Buf
                     returnEnd = consumedSegment;
 
                     // Check if we crossed _maximumSizeLow and complete backpressure
-                    long consumedBytes = new ReadOnlySequence<byte>(returnStart, _readHeadIndex, consumedSegment, consumedIndex).Length;
+                    long consumedBytes = GetLength(returnStart, _readHeadIndex, consumedSegment, consumedIndex);
                     long oldLength = _length;
                     _length -= consumedBytes;
```

Comment:
Since you just read `_length` to `oldLength` could change this to 
```csharp
_length = oldLength - consumedBytes;
```
This *might* do a second read of `_length` 
```csharp
_length -= consumedBytes;
```